src/lib/template_style/_form.scss

```diff
@@ -27,120 +35,157 @@
+    left: $radio-before-left-offset;
+  }
+  input[type='checkbox']:focus + label::before,
+  input[type='checkbox']:focus + span::before,
+  input[type='radio']:focus:checked + label::before,
+  input[type='radio']:focus:checked + span::before,
+  input[type='checkbox']:hover:checked:not(:disabled) + label::before,
+  input[type='checkbox']:hover:checked:not(:disabled) + span::before,
+  input[type='radio']:hover:checked:not(:disabled) + label::before,
+  input[type='radio']:hover:checked:not(:disabled) + span::before {
+    box-shadow: 0px 0px 2px 2px map.get(SRC.$secondary-color-palette, 100);
+  }
 
-input[type='radio']:checked + label::after,
-input[type='radio']:checked + span::after {
-  content: '';
-  display: inline-block;
-  vertical-align: middle;
-  width: 10px;
-  height: 10px;
-  border-radius: 50%;
-  background: SRC.$primary-action-color;
-  position: absolute;
-  top: 5px;
-  left: 7px;
-}
+  input[type='checkbox']:hover:not(:checked):not(:disabled) + label::before,
+  input[type='checkbox']:hover:not(:checked):not(:disabled) + span::before,
+  input[type='radio']:hover:not(:checked):not(:disabled) + label::before,
+  input[type='radio']:hover:not(:checked):not(:disabled) + span::before {
+    box-shadow: 0px 0px 2px 2px map.get(SRC.$colors, 'gray-400');
+  }
 
-input[type='radio'] + label,
-input[type='radio'] + span,
-input[type='radio'] + label,
-input[type='radio'] + span {
-  padding-left: 23px;
-  color: #515359;
-}
+  input[type='checkbox']:disabled:checked + label::before,
+  input[type='checkbox']:disabled:checked + span::before {
+    background: #ddd;
+  }
+  input[type='radio']:disabled:checked + label::after,
+  input[type='radio']:disabled:checked + span::after {
+    background: map.get(SRC.$colors, 'gray-400');
+  }
 
-.checkbox label,
-.radio label {
-  padding-left: 0px;
+  input[type='checkbox']:disabled:not(:checked) + label::before,
+  input[type='checkbox']:disabled:not(:checked) + span::before,
+  input[type='radio']:disabled + label::before,
+  input[type='radio']:disabled + span::before {
+    background: white;
+  }
+
+  input[type='checkbox']:disabled + label,
+  input[type='checkbox']:disabled + span,
+  input[type='radio']:disabled + label,
+  input[type='radio']:disabled + span {
+    color: #777;
+  }
+
+  .indeterminate-checkbox input[type='checkbox'] + label::after,
+  .indeterminate-checkbox input[type='checkbox'] + span::after {
+    content: '—';
+    color: white;
+    font-size: 12px;
+    position: absolute;
+    top: $checkbox-before-top-offset - 1px;
+    left: $checkbox-before-left-offset + 3px;
+    font-weight: bolder;
+  }
+
+  input[type='checkbox']:checked + label::after,
+  input[type='checkbox']:checked + span::after {
+    content: '';
+    transform: rotate(45deg);
+    height: 9px;
+    width: 4px;
+    border-bottom: 2px solid white;
+    border-right: 2px solid white;
+    position: absolute;
+    top: $checkbox-after-top-offset;
+    left: $checkbox-after-left-offset;
```

Comment:
Change the checkmark from unicode character (appearance depends on OS/browser) to a custom-sized div and with two adjacent borders, which happens to end up looking like the checkmark in our designs.